quantities/common/ICastOperators.cs

```diff
@@ -1,7 +1,7 @@
 namespace Quantities;
 
-public interface ICastOperators<in TSelf>
-    where TSelf : ICastOperators<TSelf>
+public interface ICastOperators<in TSelf, out TResult>
```

Comment:
This will ensure you don't have to use `Value` in the casts?

Reply:
I chose to go with casting instead of using properties, as this library is close to mathematics and physics. In that context it "feels" more natural to go with casting.
Also, and potentially more to the point, it could be argued that casting is closer to what we're doing: The `Quantity` is already "the value". Now, if we want it as a `Double` we're asking for the same thing - "the value" - in a different representation, which is precisely the semantics  of casting.